crates/sargon/src/wrapped_radix_engine_toolkit/low_level/transaction_manifest/execution_summary/transaction_manifest_execution_summary.rs

```diff
@@ -5,27 +5,41 @@ use radix_engine_toolkit::functions::manifest::execution_summary as RET_executio
 impl TransactionManifest {
     pub fn execution_summary(
         &self,
-        encoded_receipt: BagOfBytes,
+        engine_toolkit_receipt: BagOfBytes,
     ) -> Result<ExecutionSummary> {
-        let receipt: TransactionReceipt = encoded_receipt.try_into()?;
+        let network_definition = self.network_id().network_definition();
```

Comment:
oh wow, was it really this complicated? Fells like we really should be able to simplify this... is this what results in the least amount of code from Host? We should always optimize for host ofc!

Reply:
I was able to copy-pasta this from RET [repo](https://github.com/radixdlt/radix-engine-toolkit/commit/12a9ce36cc8c36c728f095b450caeb45195eb431#diff-d09f986affca4b35d7053153c834828e7d0565db366475aea8fcbf0814a60035R107-R121) 😅